lib/msal-common/src/authority/AuthorityFactory.ts

```diff
@@ -17,44 +17,71 @@ import { AdfsAuthority } from "./AdfsAuthority";
 export class AuthorityFactory {
 
     /**
-     * Parse the url and determine the type of authority
+     * Create an authority object of the correct type based on the url
+     * Performs basic authority validation - checks to see if the authority is of a valid type (i.e. aad, b2c, adfs)
+     * 
+     * Also performs endpoint discovery.
+     * 
+     * @param defaultAuthority 
+     * @param networkClient 
+     * @param authorityUri 
+     * @param adfsDisabled 
      */
-    private static detectAuthorityFromUrl(authorityString: string): AuthorityType {
-        const authorityUrl = new UrlString(authorityString);
-        const components = authorityUrl.getUrlComponents();
-        const pathSegments = components.PathSegments;
+    static async createDiscoveredInstance(authorityUri: string, networkClient: INetworkModule, adfsDisabled?: boolean): Promise<Authority> {
+        // Initialize authority and perform discovery endpoint check.
+        const acquireTokenAuthority: Authority = AuthorityFactory.createInstance(authorityUri, networkClient);
+
+        // This is temporary. Remove when ADFS is supported for browser
+        if (adfsDisabled && acquireTokenAuthority.authorityType == AuthorityType.Adfs) {
```

Comment:
Can we just get rid of this? My understanding is that the only blocker at the moment is knowing what to tell customers when they ask how to configure their ADFS servers to support Auth Code. I believe the library should support it, so I would argue we shouldn't artificially block this.

Reply:
done.